core/templates/pages/exploration-player-page/services/command-executor.service.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Oppia interaction from outer iframe with messaging commands.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import {WindowRef} from 'services/contextual/window-ref.service.ts';
+import {WindowWrapperMessageService} from
+  'pages/exploration-player-page/services/window-wrapper-message.service';
+require('pages/exploration-player-page/services/exploration-engine.service.ts');
+import {ServicesConstants} from 'services/services.constants';
+
+@Injectable({providedIn: 'root'})
+export class CommandExecutorService {
+  private commandToFunctionMap = {
+    CONTINUE: this.continueClick,
+    ENTER_TEXT_NUMBER_UNITS: this.fillTextBox,
+    ADD_SET: this.addSet,
+    REMOVE_SET: this.removeSet,
+    ENTER_FRACTION: this.enterFraction,
+    SELECT_ITEM_BULLET: this.selectItemBullet,
+    SUBMIT: this.submit,
+  };
+  setElementsOnPage = 0;
+  hostname = '';
+  cachedOuterFrameMessage = '';
+  windowWrapperMessageService = null;
+  constructor(private windowRef: WindowRef) {
+    this.setElementsOnPage = 0;
+    this.hostname = '';
+    this.cachedOuterFrameMessage = '';
+    this.windowWrapperMessageService =
+      new WindowWrapperMessageService(windowRef);
+  }
+
+  sendParentReadyState(windowRef) {
+    this.windowWrapperMessageService.postMessageToParent(
+      'Ready to receive hostname', '*');
+  }
+
+  getOuterFrameEvents(windowRef) {
```

Comment:
This isn’t really a getter—it’s not returning anything. In fact, it seems to be initializing. Can we perform all initing work in a single public function rather than in pieces? That will simplify the public contract of this class.